src/modules/esl-utils/dom/events/target.ts

```diff
@@ -15,29 +19,42 @@ export class SyntheticEventTarget implements EventTarget {
       minCount = type;
       type = 'change';
     }
-    return this._listeners.size > minCount;
+    return this._listeners[type]?.length > minCount;
   }
 
   public addEventListener(callback: EventListenerOrEventListenerObject): void;
-  public addEventListener(type: 'change', callback: EventListenerOrEventListenerObject): void;
-  public addEventListener(type: any, callback: EventListenerOrEventListenerObject = type): void {
+  public addEventListener(type: string, callback: EventListenerOrEventListenerObject): void;
+  public addEventListener(type: string | EventListenerOrEventListenerObject = 'change', callback?: EventListenerOrEventListenerObject): void {
+    if (typeof type !== 'string') {
+      callback = type;
+      type = 'change';
+    }
+
     validateEventListenerType(callback);
-    this._listeners.add(callback);
+    const listeners = this._listeners[type];
+    this.hasEventListener(type) ? listeners.push(callback!) : Object.assign(this._listeners, {[type]: [callback!]});
```

Comment:
```suggestion
    if (this._listeners[type] && !this._listeners[type].includes(callback!)) this._listeners[type].push(callback!)
    else Object.assign(this._listeners, {[type]: [callback!]});
```

Reply:
The current implementation accepts a single listener to be subscribed multiple times, but it's unacceptable as this fact may used by the consumers.